### PR TITLE
Adds async way to get value from NamedValue

### PR DESCRIPTION
**Pull request by jas-valgotar**

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Public/Values/NamedValue.cs [hunk 1]
```diff
@@ -17,7 +18,17 @@ public class NamedValue
     {
         public string Name { get; }
 
-        public FormulaValue Value => _value ?? _getFormulaValue().ConfigureAwait(false).GetAwaiter().GetResult();
+        public FormulaValue Value => ValueAsync().GetAwaiter().GetResult();
+
+        public async Task<FormulaValue> ValueAsync()
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Core/Public/Values/NamedValue.cs:23]: Should this be a property? (instead of a method)?

**Review by MikeStall — APPROVED** [review]

File: src/libraries/Microsoft.PowerFx.Core/Public/Values/NamedValue.cs (hunk 1 above)
Comment (jas-valgotar) [reply src/libraries/Microsoft.PowerFx.Core/Public/Values/NamedValue.cs:23]: C# don't support async props.
https://stackoverflow.com/questions/6602244/how-to-call-an-async-method-from-a-getter-or-setter#:~:text=There%20is%20no%20technical%20reason,be%20kicking%20off%20background%20operations.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**